### ISSUE TITLE
 How do "xnoremap p pgvy" work correctly?

### ISSUE THREAD

**Issue by okamototomoyuki1988** [jaredpar/VsVim#2260]
I want to paste the same words over and over again,
I entered "xnoremap p pgvy" in .vsvimrc.
Next, when pasting to a word shorter than the number of copied words,
When pasting next time, a part of the copied word disappears.
How will I be able to paste correctly?

**Comment by ricksladkey** [comment]
I'm not sure I fully understand what you are trying to do, but I do see a VsVim bug in there.

For your specific case, it seems to me that `p` in visual mode repeatedly would work just fine and that no mapping is needed at all because if you paste from visual mode, reselect what you just pasted and yank it, the contents of the unnamed register *shouldn't* change. In other words, you can yank once and paste multiple times.

Nevertheless, this sequence of commands should work, and it doesn't because `gv` in this case doesn't always select what was just put due the deficiencies in the way VsVim tracks changes to the previous selection.

In short, unless I am misunderstanding something, you should just use `p`, but I can confirm that `gv` after `p` in visual mode does have a bug.

**Comment by sqlkoala** [comment]
I think the problem is this: if you want to paste the same text multiple times and overwrite other text with it, then you get the problem that vim puts the overwritten text in the "register. That's why you try to "reyank" the text in your mapping.
One solution could be to paste FROM another register, the "last yank" register 0. You first yank what you want to paste, then
`"0p` in visual mode, or easier with a mapping to P:
`vnoremap P "0p`

**Comment by ricksladkey** [comment]
I fixed this and added a test for it in PR #2269.

**Comment by okamototomoyuki1988** [comment]
@sqlkoala
It worked. Thank you very much.

@ricksladkey 
Thank you for fixing.
